Lesson 6/src/simulation/Simulation.java

```diff
@@ -0,0 +1,178 @@
+package simulation;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import design.Item;
+import design.Rocket;
+import design.U1;
+import design.U2;
+
+public class Simulation {
+
+	public ArrayList<Item> loadItems(String file) throws FileNotFoundException
+	{
+		ArrayList<Item> items=new ArrayList<>();
+		File phase=new File(file);
+		Scanner scanner=new Scanner(phase);
+
+		while(scanner.hasNextLine())
+		{
+			String line=scanner.nextLine();
+			String[] lineSplit=line.split("=");
+
+			Item i=new Item(lineSplit[0], Integer.valueOf(lineSplit[1]));
+			items.add(i);
+
+
+		}
+		return items;
+
+	}
+
+	public ArrayList<Rocket> loadU1 (ArrayList<Item> getItems)
+
+	{  
+
+		ArrayList<Rocket> list_rocketsU1=new ArrayList<>();
+		Rocket r1=new U1();
+		for(Item h:getItems) 
+		{
+
+			if(r1.canCarry(h))
+			{
+
+				
+				r1.carry(h);
+
+
+			}
+			else
+			{
+
+				list_rocketsU1.add(r1);
+				r1=new U1();
+				
+				
+			}
+
+		}	
+		list_rocketsU1.add(r1);
+
+		return list_rocketsU1;
+	}
+
+	public ArrayList<Rocket> loadU2 (ArrayList <Item> getItems)
+	{
+
+		ArrayList<Rocket> 	list_rocketsU2=new ArrayList<>();
+		Rocket r2=new U2();
+
+		for(Item h:getItems) {
+
+			if(r2.canCarry(h))
+			{
+				//list_rocketsU2.add(r2);
+				r2.carry(h);
+
+
+			}
+			else
+			{
+
+				list_rocketsU2.add(r2);
+				r2=new U1();
+			
+				
+			}
+
+		}	
+		list_rocketsU2.add(r2);
+
+		return list_rocketsU2;
+
+	}
+	public int runSimulation(ArrayList <Rocket> rockets)
+
+	{
+		int rocketsExploded=0;
+		int  rocketSuccess=0;
+		int rocketsCrashed=0;
+
+		for(Rocket a1:rockets)
+		{
+			
+			while(a1.launch()== false)  
+			{
+				rocketsExploded++;
+				a1.launch();
+
+				
+			}
+			
+             while(a1.land()==false)
+			{    
+				
+				a1.launch();
```

Comment:
Same as previous comment, but also here I guesses you wanted to write `a1.land()` instead of `a1.launch()`

This line I guess is doing a land attempt but your program is not aware of the result, neither being is a success nor a failure.